## Задача 1
Оформляйте ноутбук, используя эти советы:
## Номер задачи - заголовок 2
### Номер подзадачи - заголовок 3
Предоставленные наборы данных оформляйте, как код
Рекомендации для преподавателей по оценке задания:
Смотреть, чтобы студент красиво оформлял ноутбук, использовал ячейки с текстом, указывал номера заданий

## Задача 2
**На складе лежат разные фрукты в разном количестве. Нужно написать функцию, которая на вход принимает любое количество названий фруктов и их количество, а возвращает общее количество фруктов на складе**

In [1]:
from collections import Counter

In [2]:
def Arrival_fruits(rest, **fruits):
    
    dicts = [rest, fruits]
    rest = dict(sum(map(Counter, dicts), Counter()))
    
    fruits_total = sum(rest.values())
    fruits_name = rest.keys()

    return rest, fruits_name, fruits_total

In [3]:
# Начальные остатки на складе
rest_STORE = {'Яблоки': 1200,
              'Помело': 50,
              'Апельсины': 900,
              'Ананасы': 600,
              'Груши': 1500,
              'Мандарины': 280,
              'Бананы': 800}

# Поступило фруктов
fruits_dic = {'Яблоки': 300,
             'Кокос': 100,
             'Апельсины': 450,
             'Груши': 250,
             'Мандарины': 280,
             'Слива': 170}

rest_STORE, names, total = Arrival_fruits(rest_STORE, **fruits_dic)

print(f'Всего на складе фруктов {tuple(names)} = {total} кг.')

Всего на складе фруктов ('Яблоки', 'Помело', 'Апельсины', 'Ананасы', 'Груши', 'Мандарины', 'Бананы', 'Кокос', 'Слива') = 6880 кг.


Ответ.  
Всего на складе 6880 кг фруктов.

## Задача 3
**Дан список с затратами на рекламу. Но в данных есть ошибки, некоторые затраты имеют отрицательную величину. Удалите такие значения из списка и посчитайте суммарные затраты [100, 125, -90, 345, 655, -1, 0, 200].**  
**Используйте list comprehensions**

In [4]:
def total_costs(c_price:list) -> int:
    temp = [i if i > 0 else 0 for i in costs_price]
    print(temp)
    return sum(temp)

costs_price = [100, 125, -90, 345, 655, -1, 0, 200]

print(f'Сумарные затраты на рекламу = {total_costs(costs_price)} руб.')

[100, 125, 0, 345, 655, 0, 0, 200]
Сумарные затраты на рекламу = 1425 руб.


Ответ.  
Сумарные затраты на рекламу 1425 руб.

## Задача 4
**Даны два списка:** 
1) _Дата покупки_  
['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
2) _Суммы покупок по датам_  
[1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

In [5]:
import datetime

In [6]:
date = ['2021-09-14', '2021-12-15', '2021-09-08', '2021-12-05', '2021-10-09', '2021-09-30', '2021-12-22', '2021-11-29', 
        '2021-12-24', '2021-11-26', '2021-10-27', '2021-12-18', '2021-11-09', '2021-11-23', '2021-09-27', '2021-10-02', 
        '2021-12-27', '2021-09-20', '2021-12-13', '2021-11-01', '2021-11-09', '2021-12-06', '2021-12-08', '2021-10-09', 
        '2021-10-31', '2021-09-30', '2021-11-09', '2021-12-13', '2021-10-26', '2021-12-09']
revenue = [1270, 8413, 9028, 3703, 5739, 4095, 295, 4944, 5723, 3701, 4471, 651, 7037, 4274, 6275, 4988, 6930, 2971, 
           6592, 2004, 2822, 519, 3406, 2732, 5015, 2008, 316, 6333, 5700, 2887]

### 4.1 
**Найдите, какая выручка у компании в ноябре**
*(Используйте list comprehensions)*

In [10]:
month = '11'  
revenue_november = [revenue[i] for i in range(len(date)) if date[i][5:7] == month] 
print(f'Выручка компании в ноябре составила {sum(revenue_november)} руб.')

Выручка компании в ноябре составила 25098 руб.


### 4.2 
**Найдите выручку компании в зависимости от месяца**  
_Для этого напишите функцию, которая на вход принимает список с датами и список с выручкой, а на выходе словарь, где ключи - это месяцы, а значения - это выручка._  
_Используйте аннотирование типов._

In [11]:
def get_revenue_month(date: list[str], revenue: list[int]) -> dict: # 4.2
    month = [('01','январь'), ('02','февраль'), ('03','март'), ('04','апрель'), ('05','май'),('06','июнь'),
             ('07','июль'),('08','август'),('09','сентябрь'), ('10','октябрь'),('11','ноябрь'), ('12','декабрь')]
    for i in range(len(month)):
        for j in range(len(date)):
            if month[i][0] == date[j][5:7]:
                date[j] = month[i][1]
    revenue_dict = dict()
    for i, value in enumerate(revenue):
        revenue_dict[date[i]] = revenue_dict.get(date[i], 0) + value
    return revenue_dict

In [12]:
print(f'Выручка компании по месяцам: {get_revenue_month(date, revenue)}')

Выручка компании по месяцам: {'сентябрь': 25647, 'декабрь': 45452, 'октябрь': 28645, 'ноябрь': 25098}


Ответ.  
Выручка компании в ноябре составила 25098 руб.  
Выручка компании по месяцам: 
<table align="center">
  <tr align="center">
    <td>Месяц</td>
    <td>Выручка, руб.</td>
  </tr>
  <tr align="center">
    <td>Сентябрь</td>
    <td>25647</td>
  </tr>
  <tr align="center">
    <td>Октябрь</td>
    <td>28645</td>
  </tr>
  <tr align="center">
    <td>Ноябрь</td>
    <td>25098</td>
  </tr>
  <tr align="center">
    <td>Декабрь</td>
    <td>45452</td>
  </tr>
</table>

